stubs/PDOStatement.stub

```diff
@@ -1,11 +1,18 @@
 <?php
 
 /**
- * @implements Traversable<array<int|string, mixed>>
- * @implements IteratorAggregate<array<int|string, mixed>>
+ * @template TValue
+ * @implements Traversable<int, TValue>
+ * @implements IteratorAggregate<int, TValue>
  * @link https://php.net/manual/en/class.pdostatement.php
  */
 class PDOStatement implements Traversable, IteratorAggregate
 {
-
+    /**
+     * @template T
+     * @param class-string<T> $class
+     * @param array<mixed> $ctorArgs
+     * @return false|T
+     */
+    public function fetchObject($class = \stdclass::class, array $ctorArgs = array()) {}
```

Comment:
Please test that the `fetchObject` method behaves as expected. I suspect you need to add `T of object`.

Reply:
added a test. IMO it works as expected

Reply:
> I suspect you need to add `T of object`.

good point. also submitted to psalm with https://github.com/vimeo/psalm/pull/7274